devito/finite_differences/tools.py

```diff
@@ -47,6 +46,7 @@ def adjoint(self, matvec):
 def check_input(func):
     @wraps(func)
     def wrapper(expr, *args, **kwargs):
+        from devito.finite_differences import Differentiable
```

Comment:
this is too hacky and therefore not acceptable, we need to fix the cyclic import issue differently